### PR TITLE
Release/1.1.0

### PR DESCRIPTION
**Pull request by CrepeGoat**
waiting on #56 for release

### CODE REVIEW

**Comment by CrepeGoat** [comment]
#56 merged -> ready for review!

File: endaq/calc/utils.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+from typing import Optional
+import warnings
+
+import numpy as np
+import pandas as pd
+
+
+def sample_spacing(
+    df: pd.DataFrame, convert: typing.Literal[None, "to_seconds"] = "to_seconds"
+):
```
Comment (SamRagusa) [inline endaq/calc/utils.py:15]: Should add a return type hint

File: endaq/calc/utils.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+        )
+
+    return 2 ** np.arange(
+        np.log2(init_freq),
+        np.log2(1 / dt) - 1,
+        1 / bins_per_octave,
+    )
+
+
+def to_dB(data: np.ndarray, reference: float, squared=False):
```
Comment (SamRagusa) [inline endaq/calc/utils.py:63]: Add typehint for return and 'squared' parameter

File: endaq/calc/utils.py [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+    return (10 if squared else 20) * (np.log10(data) - np.log10(reference))
+
+
+dB_refs = {
+    "SPL": 2e-5,  # Pascal
+    "audio_intensity": 1e-12,  # W/m²
+}
+
+
+def resample(df: pd.DataFrame, sample_rate: Optional[float] = None):
```
Comment (SamRagusa) [inline endaq/calc/utils.py:108]: Typehint for return

File: endaq/calc/shock.py [hunk 4]
```diff
@@ -11,71 +12,247 @@
...
-    """Calculate the relative displacement for a SDOF system."""
-    # Generate the transfer function
-    #   H(s) = L{z(t)}(s) / L{y"(t)}(s) = (1/s²)(Z(s)/Y(s))
-    # for the PDE
-    #   z" + (2ζω)z' + (ω^2)z = -y"
-    dt = (df.index[-1] - df.index[0]) / (len(df.index) - 1)
-    if isinstance(dt, (np.timedelta64, pd.Timedelta)):
-        dt = dt / np.timedelta64(1, "s")
 
+def _rel_displ_transfer_func(omega: float, damp: float = 0.0, dt: float = 1.0):
```
Comment (SamRagusa) [inline endaq/calc/shock.py:19]: Type Hint for return

File: endaq/calc/shock.py [hunk 5]
```diff
@@ -11,71 +12,247 @@
...
+    dt = utils.sample_spacing(accel)
+    tf = _rel_displ_transfer_func(omega, damp, dt)
+
+    return accel.apply(
+        functools.partial(scipy.signal.lfilter, tf.num, tf.den, axis=0),
+        raw=True,
+    )
+
+
+def _abs_accel_transfer_func(omega: float, damp: float = 0.0, dt: float = 1.0):
```
Comment (SamRagusa) [inline endaq/calc/shock.py:78]: Typehint for return

File: endaq/calc/shock.py [hunk 6]
```diff
@@ -11,71 +12,247 @@
...
+    :param damp: the damping factor used to generate the input PVSS
+    :return: a tuple of amplitudes and periods, each pair of which describes a
+        half-sine pulse
+
+    .. seealso::
+
+        `Pseudo Velocity Shock Spectrum Rules For Analysis Of Mechanical Shock, Howard A. Gaberson <https://info.endaq.com/hubfs/pvsrs_rules.pdf>`_
+    """
 
     def amp_factor(damp):
```
Comment (SamRagusa) [inline endaq/calc/shock.py:257]: Type Hinting

File: endaq/calc/stats.py [hunk 7]
```diff
@@ -46,21 +64,44 @@ def max_abs(
 
 
 def rms(
-    data: np.ndarray,
+    array: np.ndarray,
     axis: Union[None, typing.SupportsIndex, Sequence[typing.SupportsIndex]] = None,
     keepdims: bool = False,
 ):
```
Comment (SamRagusa) [inline endaq/calc/stats.py:70]: Type hint for return

**Review by SamRagusa — APPROVED** [review]
There's some missing type hinting but other than that this all looks solid

**Review by CFlaniganMide — CHANGES_REQUESTED** [review]
requirements.txt should be populated with all development related requirements.

**Comment by CFlaniganMide** [comment]
> 
> 
> requirements.txt should be populated with all development related requirements.

or deleted, either way.

**Review by StokesMIDE — APPROVED** [review]
As @SamRagusa noted, there were some methods missing type hints, but these were mostly on 'private' methods (and not an exposed part of the API), so I don't think they're critical. At least not enough to hold up release, imho.

File: endaq/calc/utils.py (hunk 1 above)
Comment (CrepeGoat) [reply endaq/calc/utils.py]: see 06f2b0ae993894190bbe6c5aa510cddf75c58bde

File: endaq/calc/utils.py (hunk 2 above)
Comment (CrepeGoat) [reply endaq/calc/utils.py]: see 06f2b0ae993894190bbe6c5aa510cddf75c58bde

File: endaq/calc/utils.py (hunk 3 above)
Comment (CrepeGoat) [reply endaq/calc/utils.py]: see 06f2b0ae993894190bbe6c5aa510cddf75c58bde

File: endaq/calc/shock.py (hunk 4 above)
Comment (CrepeGoat) [reply endaq/calc/shock.py]: see 06f2b0ae993894190bbe6c5aa510cddf75c58bde

File: endaq/calc/shock.py (hunk 5 above)
Comment (CrepeGoat) [reply endaq/calc/shock.py]: see 06f2b0ae993894190bbe6c5aa510cddf75c58bde

File: endaq/calc/stats.py (hunk 7 above)
Comment (CrepeGoat) [reply endaq/calc/stats.py]: see 06f2b0ae993894190bbe6c5aa510cddf75c58bde

**Review by StokesMIDE — APPROVED** [review]

**Review by SamRagusa — APPROVED** [review]

**Review by CFlaniganMide — APPROVED** [review]

**Pull request merged**